src/logger.py

```diff
@@ -0,0 +1,108 @@
+import json
+import os
+import time
+import logging
+import datetime
+from enum import Enum
+from abc import ABC
+
+
+class BaseFormatter(ABC, logging.Formatter):
+
+    def formatTime(self, record):
+        """
+        Return the creation time of the specified LogRecord as formatted text.
+        This method should be called from format() by a formatter which
+        wants to make use of a formatted time.
+        It provides an ISO 8601 format with milliseconds.
+        """
+        ct = self.converter(record.created)
+        ct_with_ms = time.mktime(ct) + (record.msecs / 1000)
+        ct_full = datetime.datetime.fromtimestamp(ct_with_ms)
+        return ct_full.isoformat()
+
+    def get_fields(self, record):
+        """
+        Return a dict wth the following fields:
+         - time: ISO 8601 format with milliseconds (from datetime.isoFormat)
+         - level: Log level in uppercase letter
+         - msg: the log message
+         - exception: exception if the logger is specifying exc_info=True
+         - stack: stack information if the logger is specifying stack_info=True
+        """
+        fields = dict()
+        fields['time'] = self.formatTime(record)
+        fields['level'] = record.levelname
+        fields['msg'] = record.getMessage()
+        if record.exc_info:
+            fields['exception'] = self.formatException(record.exc_info)
+        if record.stack_info:
+            fields['stack'] = self.formatStack(record.exc_info)
+        return fields
+
+
+class LogfmtFormatter(BaseFormatter):
```

Comment:
I think configuring the [default log formatter](https://docs.python.org/3/library/logging.html#formatter-objects) with suitable format strings should be good enough. I don't see why we need a custom formatter.

Reply:
Here is a custom formatter to use LogFmt fomat (key and value). 

We can also use a format string like `time=%(asctime)s level=%(levelname)s msg=%(message)s` .

I centralize the log fields  in `BaseFormatter` in order to use the same for JsonFormatter and LogFmtFormatter but I can also revisit that to simplify things.